docs/user-guide/ffi.md

```diff
@@ -0,0 +1,122 @@
+<script type="py">
+  from pyscript.ffi import to_js
+
+  # always default into JS {"a": 1, "b": 2}
+  to_js({"a": 1, "b": 2})
+</script>
+```
+
+!!! Note
+
+    It is still possible to specify a different `dict_converter` or use Pyodide
+    specific features while converting Python references by simply overriding
+    the explicit field for `dict_converter`. However, we cannot guarantee
+    all fields and features provided by Pyodide will work the same on MicroPython.
+
+## create_proxy
+
+In [Pyodide project](https://pyodide.org/en/stable/usage/api/python-api/ffi.html#pyodide.ffi.create_proxy), this utility guarantees that a Python lambda or callback won't get Garbage Collected immediately after.
+
+There are, still in *Pyodide*, dozen of ad-hoc utilities that work around that implementation detail, such as `create_once_callable` or `pyodide.ffi.wrappers.add_event_listener` or `pyodide.ffi.wrappers.set_timeout` and others, all methods which goal is to automatically handle the lifetime of the passed callback.
+
+There are also implicit helpers like in `Promise` where `.then` or `.catch` callbacks are implicitly handled a part to guarantee no leaks once executed.
+
+Because the amount of details could be easily overwhelming, we decided to provide an `experimental_create_proxy = "auto"` configuration option that should never require our users to care about all these details while all the generic APIs in the *JS* world should "*just work*".
+
+This flag is strictly coupled with the *JS* garbage collector and it will eventually destroy all proxies that were previously created through the [FinalizationRegistry](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/FinalizationRegistry) native utility.
+
+This flag also won't affect *MicroPython* because it rarely needs a `create_proxy` at all, when Python functions are passed to JS, but there are still cases where that conversion might be needed or explicit.
+
+For these reasons we expose also the `create_proxy` utility which does not change much in the *MicroPython* world or code, but it might be still needed in the *Pyodide* runtime.
+
+The main difference with these two different runtime is that *MicroPython* doesn't provide (yet) a way to explicitly destroy the proxy reference, while in *Pyodide* it's still desirable to explicitly invoke that `proxy.destroy()` when the function is not needed/called anymore.
+
+```html title="A classic Pyodide failure VS MicroPython"
+<!-- Throws:
+Uncaught Error: This borrowed proxy was automatically destroyed
+at the end of a function call. Try using create_proxy or create_once_callable.
+-->
+<script type="py">
+    import js
+    js.setTimeout(lambda x: print(x), 1000, "fail");
+</script>
+
+<!-- logs "success" after a second -->
+<script type="mpy">
+    import js
+    js.setTimeout(lambda x: print(x), 1000, "success");
+</script>
+```
+
+To fix the Pyodide's issue we can use the experimental flag:
```

Comment:
"To address the difference in Pyodide's behaviour, we can use the experimental flag:"